ActiveAdmin Gem

thomasjacobjr opened the pull request:
Install and Implement ActiveAdmin Gem

thomasjacobjr committed:
ActiveAdmin Gem

thomasjacobjr committed:
Migrated and configured active_admin

thomasjacobjr committed:
Various attempts to implement ActiveAdmin (in Progress)

thomasjacobjr committed:
Admin user can access dashboard

thomasjacobjr committed:
Cleanup -- admin boolean column on user table, reseeded db with admin user, dashboard working

thomasjacobjr committed:
Fixed alert message for non-admin users

thomasjacobjr committed:
Simpler attribute table ActiveAdmin User Resource

thomasjacobjr committed:
ActiveAdmin Filter for User resource set to Email only

thomasjacobjr committed:
Admin can create new user with email and admin status

thomasjacobjr committed:
Admin can successfully add a new user with password, but redirection is still wrong

thomasjacobjr committed:
Admin can delete a user from dashboard using batch actions

chikamichi commented on app/views/layouts/application.html.erb:
```diff
@@ -8,6 +8,7 @@
   <%= csrf_meta_tags %>
 </head>
 <body>
+  <p class="alert"><%= alert %>
```
Even though browsers will handle it, you could add the `</p>` closing tag here.

chikamichi commented on app/admin/user.rb:
```diff
@@ -4,7 +4,12 @@
+    column :email
+    column :admin
+    column :created_at
+  end
```
:+1:

chikamichi replied:
You should add a method call `actions` after your last `column` method call, that will take care of creating a couple links to view, edit and delete a `User` resource. It's documented in [_Customizing the Index Page_](http://activeadmin.info/docs/3-index-pages/index-as-table.html).

chikamichi commented on app/admin/user.rb:
```diff
@@ -1,9 +1,10 @@
 ActiveAdmin.register User do
-
+  filter :email 
```
:+1:

chikamichi commented on app/admin/user.rb:
```diff
@@ -1,5 +1,13 @@
+      input :admin
+      actions 
+    end
+  end
```
:+1:

chikamichi commented on app/admin/user.rb:
```diff
@@ -22,6 +22,7 @@
     column :admin
     column :created_at
   end
+
```
This commit seems empty, besides this new line.

chikamichi commented on app/admin/user.rb:
```diff
@@ -1,12 +1,15 @@
+      f.input :password
+      f.input :password_confirmation
     end
+    f.actions
```
Cool.

ActiveAdmin has you use this single `form` method to compose both the "create" and the "edit" forms. Although similar, when creating a new user you'll want to specify a `password` / `password_confirmation` but it shall not be modified in the edit form, so you'd like to hide/remove those two inputs.

You can check whether the form is in "create" or "edit" mode by checking whether the `User` resource that the form handles is an existing record or not: `f.object.new_record?` (where `f.object` is a reference to the `User` instance being either created or modified). There is an example in [the documentation about the _Customizing the Form_](http://activeadmin.info/docs/5-forms.html).

chikamichi committed:
Redirect to the previous url upon signing in

Mostly useful when accessing /admin
@see
https://github.com/plataformatec/devise/wiki/How-To:-Redirect-back-to-current-page-after-sign-in,-sign-out,-sign-up,-update#devise-40

thomasjacobjr committed:
Quick fixes

thomasjacobjr committed:
Merge pull request #12 from thomasjacobjr/devise-redirection

Redirect to the previous url upon signing in

thomasjacobjr committed:
INstalled Rails DB Viewer Gem into development

thomasjacobjr committed:
Updated Readme

thomasjacobjr committed:
Registered Page model as an ActiveAdmin resource

thomasjacobjr committed:
Configured page.rb ActiveAdmin with editable params and friendly_id support

thomasjacobjr committed:
Moved activeadmin stylesheets to vendor folder // added view on site button

thomasjacobjr committed:
Removed unnecessary CUD actions in page controller and buttons from views

thomasjacobjr committed:
Turned off confirm email for admin user creation

chikamichi commented on app/admin/page.rb:
```diff
@@ -17,6 +17,10 @@ def find_resource
+  action_item :view, only: :show do
+    link_to 'View on site', page_path(page)
+  end
+
```
:+1:

chikamichi commented:
Excellent work!

Just for information, here's what I got for app/admin/page.rb at this stage:

``` ruby
ActiveAdmin.register Page do
  controller do
    def find_resource
      scoped_collection.friendly.find(params[:id])
    end
  end

  permit_params :title, :menu_title, :body, :slug

  index do
    selectable_column
    column t('admin.page.slug'), :slug
    column :title
    column :menu_title
    column :body
    actions dropdown: true do |page|
      item t('admin.view_on_site'), page_path(page)
    end
    column :created_at
    column :updated_at
  end

  form do |f|
    f.inputs do
      f.input :title
      f.input :menu_title
      f.input :body
      f.input :slug
    end
    f.actions
  end

  action_item :view_on_site, only: :show do
    link_to t('admin.view_on_site'), page_path(page)
  end
end
```

The only differences are the support of `:slug`, and a "View on site" link in the index.
